src/metabase/api/dashboard.clj

```diff
@@ -784,38 +786,51 @@
   "How many results to return when chain filtering"
   1000)
 
-(mu/defn ^:private mappings->field-ids :- [:maybe [:set ms/PositiveInt]]
-  [parameter-mappings :- [:maybe [:or [:set dashboard-card/ParamMapping] [:sequential dashboard-card/ParamMapping]]]]
-  (set (for [{{:keys [card]} :dashcard :keys [target]} parameter-mappings
-             :let  [field-clause (params/param-target->field-clause target card)]
-             :when field-clause
-             :let  [{:keys [result_metadata]} card
-                    field-id (or
-                              ;; Get the field id from the field-clause if it contains it. This is the common case for
-                              ;; mbql queries.
-                              (mbql.u/match-one field-clause [:field (id :guard integer?) _] id)
-                              ;; Attempt to get the field clause from the model metadata corresponding to the field.
-                              ;; This is the common case for native queries in which mappings from original columns
-                              ;; have been performed using model metadata.
-                              (:id (qp.util/field->field-info field-clause result_metadata)))]
-             :when field-id]
-         field-id)))
-
-(defn- param-key->field-ids
-  "Get Field ID(s) associated with a parameter in a Dashboard.
-
-    (param-key->field-ids (t2/select-one Dashboard :id 62) \"ee876336\")
-    ;; -> #{276}"
-  [dashboard param-key]
-  {:pre [(string? param-key)]}
-  (let [{:keys [resolved-params]} (t2/hydrate dashboard :resolved-params)
-        param                     (get resolved-params param-key)]
-    (mappings->field-ids (:mappings param))))
-
-(defn- chain-filter-constraints [dashboard constraint-param-key->value]
-  (into {} (for [[param-key value] constraint-param-key->value
-                 field-id          (param-key->field-ids dashboard param-key)]
-             [field-id value])))
+(defn- find-template-tag
+  "Fetch the `:field` clause from `dashcard` referenced by `template-tag`.
+
+    (find-template-tag [:template-tag :company] some-dashcard) ; -> [:field 100 nil]"
+  [dimension card]
+  (when-let [tag (first (mbql.u/get-clause :template-tag dimension))]
+    (get-in card [:dataset_query :native :template-tags (u/qualified-name tag)])))
+
+(defn- param->op [{:keys [type]}]
```

Comment:
this function is only used once and it's quite trivial IMO, so how about we make it inline to make avoid jumping back and forth when reading `param->fields`?

if not then I suggest we rename this to `param-type->op`, it's simple enough so we don't have to jump to it.

Reply:
Okay, `param-type->op` is nice. I extract it since while it's small, it adds to the noise in the function and ideally I'd love to see a more structured way to get from a parameter type to an operator: relying on keyword names feels somewhat error-prone to me.

Reply:
done in 39ec5a256ca0cd926f3cbad9232029374e1191fe